README.md

```diff
@@ -52,6 +52,12 @@ RSP microcode can also be recompiled with this tool. Currently there is no suppo
 ## Building
 This project can be built with CMake 3.20 or above and a C++ compiler that supports C++20. This repo uses git submodules, so be sure to clone recursively (`git clone --recurse-submodules`) or initialize submodules recursively after cloning (`git submodule update --init --recursive`). From there, building is identical to any other cmake project, e.g. run `cmake` in the target build folder and point it at the root of this repo, then run `cmake --build .` from that target folder.
 
+## Nix
+If you are using nixos or have [nix installed](https://nixos.org/)
+This project can be built and run using nix flakes, to spawn a shell with both, N64Recomp and RSPRecomp programs available you can perform the command:
+
+`nix shell 'git+https://github.com/Mr-Wiseguy/N64Recomp?submodules=1#N64Recomp`
```

Comment:
Would `nix shell github:Mr-Wiseguy/N64Recomp` be OK instead? (Could be totally wrong here, I'm still new to Nix and especially flakes)

Reply:
Yes and no, 

the problem is that this repository has submodules, when using submodules, you have to specify the submodules=1, and that's not supported with all uri notations, hence you have to use the full url + ?submodules=1.

If we packaged this for nixpkgs, or forced the package to be in an specific commit (instead of the latest at the time) then we could do it your way, since the submodule resolution would be on the src attribute in the mkDerivaton.